src/components/ERACommons.jsx

```diff
@@ -94,55 +97,64 @@ export default function ERACommons(props) {
   const redirectToECMAuthUrl = async () => {
     const origin = window.location.origin;
     const redirectTo = origin + '/' + destination;
-    const authUrl = await AuthenticateNIH.getECMProviderAuthUrl(origin, redirectTo);
-    console.log('authUrl', authUrl);
-    window.location.href = authUrl;
+    try {
+      window.location.href = await AuthenticateNIH.getECMProviderAuthUrl(origin, redirectTo);
+    } catch (error) {
+      const errorMessage = 'Error from Authentication Provider: ' + error?.response?.data?.message + ': ' + currentUser.email;
+      displayError(errorMessage);
+    }
   };
 
   const deleteNihAccount = async () => {
-    const deleteResponse = await AuthenticateNIH.deleteAccountLinkage();
-    if (deleteResponse) {
+    try {
+      await AuthenticateNIH.deleteAccountLinkage();
       const response = await User.getMe();
       const eraAuthState = extractEraAuthenticationState(response.properties);
       setAuthorized(eraAuthState.isAuthorized);
       setExpirationCount(eraAuthState.expirationCount);
       setEraCommonsId(undefined);
       onNihStatusUpdate(eraAuthState.nihValid);
       setSearch('');
-    } else {
-      setNihError(true);
-      document.getElementById('era-commons-id').scrollIntoView({block: 'start', inline: 'nearest', behavior: 'smooth'});
+    } catch (_error) {
+      const errorMessage = 'Something went wrong removing your account, please submit an error report using the "Contact Us" form.';
+      displayError(errorMessage);
     }
   };
 
+  const displayError = (message) => {
+    setNihErrorMessage(message);
+    setNihError(true);
```

Comment:
This is fine, but if `nihError` is only `true` when `nihErrorMessage` is defined, we don't need both values. And in fact I think it's better to write `{nihErrorMessage && ... {nihErrorMessage} ...}` than `{nihError && ... {nihErrorMessage} ...}`.

Reply:
Good point, I can clean that up.